src/app/clusters/operational-state-server/operational-state-delegate.h

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+constexpr size_t kErrorStateLabelMaxSize       = 64u;
+constexpr size_t kErrorStateDetailsMaxSize     = 64u;
+constexpr size_t kOperationalStateLabelMaxSize = 64u;
+
+struct OperationalErrorStateStruct
```

Comment:
Please document why this exists and why it's distinct from `ErrorStateStruct`,  Similar for OperationalStateStruct, which is not the same as `Clusters::OperationalState::Structs::OperationalStateStruct` (which is _very_ confusing).

Perhaps these types should have different names that would make it clearer how they are supposed to be used?

Reply:
Done.